lighthouse-core/test/audits/long-tasks-test.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const LongTasks = require('../../audits/long-tasks.js');
+
+const acceptableTrace = require('../fixtures/traces/progressive-app-m60.json');
+const devtoolsLog = require('../fixtures/traces/progressive-app-m60.devtools.log.json');
+
+/* eslint-env jest */
+
+describe('Long tasks audit', () => {
+  it('should filter out top level tasks shorter than 50 ms', async () => {
+    const artifacts = {
+      traces: {defaultPass: acceptableTrace},
+      devtoolsLogs: {defaultPass: devtoolsLog},
+    };
+    const result = await LongTasks.audit(artifacts, {computedCache: new Map()});
+    expect(result.details.items).toHaveLength(4);
+    expect(result.displayValue).toBeDisplayString('4 long tasks found');
+
+    for (const item of result.details.items) {
+      expect(Number.isFinite(item.start)).toBeTruthy();
+      expect(Number.isFinite(item.duration)).toBeTruthy();
+      expect(item.duration).toBeGreaterThanOrEqual(50);
+    }
+  });
```

Comment:
did y'all discuss more tests? It would be good to hit a few more things in there, unbounded tasks, unattributable URLs, etc. Those situations may already be in our existing fixtures (but will take a little digging to make sure they're in there), and there's [`create-test-trace.js`](https://github.com/GoogleChrome/lighthouse/blob/aaa6f7455cc415151759c52e741becac97a2ccde/lighthouse-core/test/create-test-trace.js) which can help make a simple valid-ish trace with certain top-level task characteristics.

Also, smoke test(s)! They might be tricky to avoid flakiness, but there are enough intentionally constructed perf/metric tests in there that at least some things are stable enough to assert.

Reply:
+1 to more test cases around here, specifically around the differences between provided/devtools throttling and simulated as well as something that would catch the fact that simulated tasks need to be multiplied *before* they're filtered out.

for non-flaky smokes, you should be able to manually construct particularly long, long tasks and then assert the first N are *at least* as long as you expect them. poor CI will probably create a few more on their own, but they should hopefully be just above the threshold and not creep into the longest long tasks positions